File: backend/streetlight/src/main/java/com/kryeit/auth/inventory/Item.java
package com.kryeit.auth.inventory;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public class Item {
    private final String id;
    private final String name;
    private final int width;
    private final int height;
    private final Map<String, Object> properties = new HashMap<>();

    public Item(String id, String name, int width, int height) {
        this.id = id;
        this.name = name;
        this.width = width;
        this.height = height;
    }

    public String getId() { return id; }
    public String getName() { return name; }
    public int getWidth() { return width; }
    public int getHeight() { return height; }

    public void setProperty(String key, Object value) {
        properties.put(key, value);
    }

    public Object getProperty(String key) {
        return properties.get(key);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Item item = (Item) o;
        return Objects.equals(id, item.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: frontend/bins/src/components/ui/inventory/models/Item.js
export default class Item {
    constructor(id, name, width, height, imageUrl = null) {
        this.id = id;
        this.name = name;
        this.width = width;
        this.height = height;
        this.imageUrl = imageUrl;
        this.properties = {};
    }

    setProperty(key, value) {
        this.properties[key] = value;
        return this;
    }

    getProperty(key) {
        return this.properties[key];
    }

    toJSON() {
        return {
            id: this.id,
            name: this.name,
            width: this.width,
            height: this.height,
            imageUrl: this.imageUrl,
            properties: {...this.properties}
        };
    }

    static fromJSON(json) {
        const item = new Item(
            json.id,
            json.name,
            json.width || 1,
            json.height || 1,
            json.imageUrl
        );

        if (json.properties) {
            item.properties = {...json.properties};
        }

        return item;
    }
}